src/core/mempool.rs

```diff
@@ -1009,17 +1009,25 @@ impl MemPoolDB {
     ///  highest-fee-first order.  This method is interruptable -- in the `settings` struct, the
     ///  caller may choose how long to spend iterating before this method stops.
     ///
-    ///  `todo` returns a boolean representing whether or not to keep iterating.
+    ///  `todo` returns an option to a `TransactionEvent` representing the outcome, or None if we
+    ///  hit an error that wasn't transaction specific.
+    ///
+    /// `output_events` is modified in place, adding all substantive transaction events output by `todo`.
     pub fn iterate_candidates<F, E, C>(
         &mut self,
         clarity_tx: &mut C,
+        output_events: &mut Vec<TransactionEvent>,
```

Comment:
Can you add a rustdoc for this new argument? Also, passing `&mut Vec<T>` or `&Vec<T>` should be avoided, in almost all cases, it's preferable to use `&mut [T]` or `&[T]`, because it gives you more flexibility in the caller.

Reply:
1) I added to the comments for this.

2) I think it has to be a `Vec` because of the `push`:

```
error[E0599]: no method named `push` found for mutable reference `&mut [TransactionEvent]` in the current scope
    --> src/core/mempool.rs:1093:43
     |
1093 | ...                   output_events.push(tx_event);
     |                                     ^^^^ method not found in `&mut [TransactionEvent]`
```

There actually isn't that much latitude for the caller.